src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -189,7 +184,9 @@ public function normalize($data, string $format = null, array $context = [])
      */
     public function denormalize($data, string $type, string $format = null, array $context = [])
     {
-        if (isset(self::SCALAR_TYPES[$type])) {
+        if (isset(ScalarDenormalizer::SCALAR_TYPES[$type]) && !$this->hasScalarDenormalizer()) {
+            trigger_deprecation('symfony/serializer', '5.2', 'Denormalizing scalar values without registering the "%s" is deprecated.', ScalarDenormalizer::class);
```

Comment:
SemVer forbids deprecating things in patch releases so this PR should be merged into branch `5.2` if it's still ok, `5.x` otherwise.

Reply:
Yes, not even sure the implementation will be accepted.
That's way I tagged the deprecation on `5.2`, the milestone will be change later if this foes any further.